src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Usage/CSharpImplementGenericMathInterfacesCorrectly.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.NetCore.Analyzers.Usage;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Usage
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpImplementGenericMathInterfacesCorrectly : ImplementGenericMathInterfacesCorrectly
+    {
+        protected override SyntaxNode? FindTheTypeArgumentOfTheInterfaceFromTypeDeclaration(ISymbol typeSymbol, ISymbol anInterfaceSymbol)
+        {
+            foreach (SyntaxReference syntaxReference in typeSymbol.DeclaringSyntaxReferences)
+            {
+                SyntaxNode typeDefinition = syntaxReference.GetSyntax();
+                if (typeDefinition is BaseTypeDeclarationSyntax baseType)
+                {
+                    return FindTypeArgumentFromBaseInterfaceList(baseType.BaseList.Types, anInterfaceSymbol);
+                }
+            }
+
```

Comment:
If `FindTypeArgumentFromBaseInterfaceList` returned null, shouldn't we continue looking for the rest of `DeclaringSyntaxReferences`? ie, adjust the condition to the following?

```csharp
if (typeDefinition is BaseTypeDeclarationSyntax baseType &&
    FindTypeArgumentFromBaseInterfaceList(baseType.BaseList.Types, anInterfaceSymbol) is { } node)
{
    return node;
}
```

Reply:
Good point, `DeclaringSyntaxReferences` seems it could have multiple SyntaxReferences only if it was partial type with different physical sources, not sure if it possible to add different sources in a unit test, so did not add a test case, but applied your comment